src/pylife/materialdata/woehler/fatigue_data.py

```diff
@@ -142,6 +142,31 @@ def conservative_fatigue_limit(self):
 
         return self
 
+    def set_fatigue_limit(self, fatigue_limit=None):
```

Comment:
Why the `None` default? I would assume that whoever calls the method wants to set it. Allowing `None` makes the whole function way more complicated. Do we need it?

Reply:
We don't "really" need it. I assumed people might forget to add a value. I guess it is better to get rid of it.